File: solutions/ts/11.ts
function decodeFilename(filename: string): string {
  const filePartStartIndex = filename.indexOf("_") + 1;
  const filePart = filename.slice(filePartStartIndex);
  const fileNameLastIndex = filePart.lastIndexOf(".");
  return filePart.slice(0, fileNameLastIndex);
}

const file1 = decodeFilename("2023122512345678_sleighDesign.png.grinchwa");
console.log(file1);
// ➞ "sleighDesign.png"

const file2 = decodeFilename("42_chimney_dimensions.pdf.hack2023");
console.log(file2);
// ➞ "chimney_dimensions.pdf"

const file3 = decodeFilename("987654321_elf-roster.csv.tempfile");
console.log(file3);
// ➞ "elf-roster.csv"

/* 
  Score: ★★★★★
*/

File: solutions/python/11.py
def decode_filename(filename: str) -> str:
    file_part_start_index = filename.find("_") + 1
    file_part = filename[file_part_start_index:]
    file_name_last_index = file_part.rfind(".")
    return file_part[:file_name_last_index]


if __name__ == "__main__":
    file1 = decode_filename("2023122512345678_sleighDesign.png.grinchwa")
    print(file1)
    # ➞ "sleighDesign.png"

    file2 = decode_filename("42_chimney_dimensions.pdf.hack2023")
    print(file2)
    # ➞ "chimney_dimensions.pdf"

    file3 = decode_filename("987654321_elf-roster.csv.tempfile")
    print(file3)
    # ➞ "elf-roster.csv"

"""
  Score: ★★★★★
"""
